components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/APIManagerTenantSyncListener.java

```diff
@@ -0,0 +1,371 @@
+                // set username only it is a creation request
+                owner.setUsername(tenantInfo.getAdmin());
+            }
+            owner.setPassword(tenantInfo.getAdminPassword());
+            owner.setEmail(tenantInfo.getEmail());
+            owner.setFirstname(tenantInfo.getFirstname());
+            owner.setLastname(tenantInfo.getLastname());
+            owners.add(owner);
+            tenant.setOwners(owners);
+        }
+
+        if (TenantManagement.EVENT_ACTIVATE_TENANT_URI.equals(eventURI)) {
+            TenantManagementEventDTO.LifecycleStatus lifecycleStatus = new TenantManagementEventDTO.LifecycleStatus();
+            lifecycleStatus.setActivated(tenantInfo.isActive());
+            tenant.setLifecycleStatus(lifecycleStatus);
+        }
+
+        createEventDetail.setTenant(tenant);
+        events.put(eventURI, createEventDetail);
+        eventDTO.setEvents(events);
+        return eventDTO;
+    }
+
+    /**
+     * Runnable Thread to send Event
+     */
+    public static class EventRunner implements Runnable {
+
+        private String notificationEndpoint;
+        private String username;
+        private String password;
+        private Map<String, String> headers;
+        private TenantManagementEventDTO event;
+
+        public EventRunner(String notificationEndpoint, String username, String password, Map<String, String> headers,
+                TenantManagementEventDTO event) {
+
+            this.notificationEndpoint = notificationEndpoint;
+            this.username = username;
+            this.password = password;
+            this.headers = headers;
+            this.event = event;
+        }
+
+        @Override
+        public void run() {
+            
+            if (log.isDebugEnabled()) {
+                log.debug("Sending HTTP request to notification endpoint: " + notificationEndpoint);
+            }
+
+            try (CloseableHttpClient httpClient = HTTPClientUtils.createClientWithCustomHostnameVerifier().build()) {
+
+                HttpPost httpPost = new HttpPost(notificationEndpoint);
+                if (StringUtils.isNotEmpty(username) && StringUtils.isNotEmpty(password)) {
+                    byte[] credentials = Base64
+                            .encodeBase64((username + ":" + password).getBytes(StandardCharsets.UTF_8));
+                    httpPost.addHeader("Authorization", "Basic " + new String(credentials, StandardCharsets.UTF_8));
+                }
+
+                headers.forEach((key, value) -> {
+                    httpPost.addHeader(key, value);
+                });
+
+                String content = new Gson().toJson(event);
+                httpPost.setEntity(new StringEntity(content, ContentType.APPLICATION_JSON));
+                httpClient.execute(httpPost, response -> {
+                    if (response.getCode() != HttpStatus.SC_OK) {
+                        log.error("Error while notifying API Manger for tenant creation. Status code: "
```

Comment:
This should be a generic implementation, not specific to API Manager, so please avoid that term.

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/7075/commits/a6be893041c5d4a898b3d2f74d8525409740cf5c